models/user_avatar.go

```diff
@@ -60,61 +60,45 @@ func (u *User) generateRandomAvatar(e db.Engine) error {
 	return nil
 }
 
-// SizedRelAvatarLink returns a link to the user's avatar via
-// the local explore page. Function returns immediately.
-// When applicable, the link is for an avatar of the indicated size (in pixels).
-func (u *User) SizedRelAvatarLink(size int) string {
-	return setting.AppSubURL + "/user/avatar/" + u.Name + "/" + strconv.Itoa(size)
-}
-
-// RealSizedAvatarLink returns a link to the user's avatar. When
-// applicable, the link is for an avatar of the indicated size (in pixels).
-//
-// This function make take time to return when federated avatars
-// are in use, due to a DNS lookup need
-//
-func (u *User) RealSizedAvatarLink(size int) string {
+// AvatarLinkWithSize returns a link to the user's avatar with size
+func (u *User) AvatarLinkWithSize(size int) string {
 	if u.ID == -1 {
-		return DefaultAvatarLink()
+		// ghost user
+		return avatars.DefaultAvatarLink()
 	}
 
+	var useLocalAvatar bool
+
 	switch {
 	case u.UseCustomAvatar:
-		if u.Avatar == "" {
-			return DefaultAvatarLink()
-		}
-		if size > 0 {
-			return setting.AppSubURL + "/avatars/" + u.Avatar + "?size=" + strconv.Itoa(size)
-		}
-		return setting.AppSubURL + "/avatars/" + u.Avatar
+		useLocalAvatar = true
 	case setting.DisableGravatar, setting.OfflineMode:
+		useLocalAvatar = true
 		if u.Avatar == "" {
 			if err := u.GenerateRandomAvatar(); err != nil {
 				log.Error("GenerateRandomAvatar: %v", err)
 			}
 		}
+	default:
+		useLocalAvatar = false
+	}
+
+	if useLocalAvatar {
+		if u.Avatar == "" {
+			return avatars.DefaultAvatarLink()
+		}
 		if size > 0 {
```

Comment:
Does something speak against
```suggestion
		else if size > 0 {
```
here?

Reply:
The `else` doesn't help about the logic.  And the `size` and `u.Avatar` are independent variables, it's better to keep them in different `if`.